src/modules/fancyzones/editor/FancyZonesEditor/GridEditor.xaml.cs

```diff
@@ -40,27 +40,27 @@ public GridEditor()
         private void GridEditor_Loaded(object sender, RoutedEventArgs e)
         {
             GridLayoutModel model = (GridLayoutModel)DataContext;
-            if (model != null)
+            if (model == null)
             {
-                _data = new GridData(model);
-                _dragHandles = new GridDragHandles(AdornerLayer.Children, Resizer_DragDelta, Resizer_DragCompleted);
-
-                int zoneCount = _data.ZoneCount;
-                for (int i = 0; i <= zoneCount; i++)
-                {
-                    AddZone();
-                }
+                return;
             }
 
+            _data = new GridData(model);
+            _dragHandles = new GridDragHandles(AdornerLayer.Children, Resizer_DragDelta, Resizer_DragCompleted);
+            _dragHandles.InitDragHandles(model);
+
             Model = model;
-            if (Model == null)
+            Model.PropertyChanged += OnGridDimensionsChanged;
+
+            int zoneCount = _data.ZoneCount;
+            for (int i = 0; i <= zoneCount; i++)
```

Comment:
shouldn't we do `i < zoneCount` instead?

Reply:
Yes, but with a little fix in `GridData`. In fact, `ZoneCount` represents the biggest used index in the grid, so the name is misleading. I'll update the code so that it would represent exactly what it means.